### DISCUSSION TITLE
Add Java JDK 21 to LTS versions in "workarounds:javaLTSVersions"

### DISCUSSION THREAD

**Discussion by jpgoelz** [renovatebot/renovate#25168] [Suggest an Idea]
### Tell us more.

Hi all!

As of now, `"workarounds:javaLTSVersions"` only includes Java JDKs 8, 11 and 17 (see [Code](https://github.com/renovatebot/renovate/blob/8432d3708c57851e9f0c58a39be4495f4bd87aa5/lib/config/presets/internal/workarounds.ts#L108C54-L108C54)).

Java JDK 21, which has been released on September 19, is supposedly an LTS version and should be added to the `allowedVersions` Regex.

What are your thoughts?

**Comment by rarkins** [comment]
https://inside.java/2023/07/06/java-21-is-no-lts/

**Comment by jpgoelz** [comment]
Thank you, Rhys for the link!

I understand what you are trying to convey but in the video you posted.
The Video talks about, how Java 21 is not anything that can get support, but the different vendors can. This was already clear to me. But the same is true for all Java "versions", including 17 and 11, as Nicolai himself says (at 0:43 min):
> But that doesn't make Java 21 a _long term support version_, just like 17 and 11 weren't LTS versions.

But both are part of the Regex. 

Now I am wondering:
1. Does this mean, 21 will not be added and Java will never be updated, unless we `"ignorePresets": ["workarounds:javaLTSVersions"]`?
2. What is `"workarounds:javaLTSVersions"` even about? It does not differentiate between vendors, right? So if 21 cannot be added, why was that possible for 17 and 11?

**Reply by rarkins** [reply]
This is probably a better reference and confirms we should include 21: https://www.oracle.com/java/technologies/java-se-support-roadmap.html#:~:text=Java%20SE%208%2C%2011%2C%2017,Java%2021%20in%20September%202023.

PR welcome

**Reply by jpgoelz** [reply]
I'll get right to it. 🥳

**Reply by jpgoelz** [reply]
After a run-in with the CLAassistant (I had my work mail address configured 🤦) here it is: https://github.com/renovatebot/renovate/pull/25175.
